ai_replica/utils/system.py

```diff
@@ -0,0 +1,73 @@
+"""Tools for environment- and OS-related stuff"""
+
+import os
+import platform
+import subprocess
+import webbrowser
+
+
+def is_expected_python_version(expected_version):
+    """Returns True if the actual Python version is the same as the expected version, and False otherwise.
+
+    It will also return True if the actual version is a subset of the expected version. For example:
+
+    | expected | actual  | result |
+    | -------- | ------- | ------ |
+    | "3.8.1"  | "3.8.1" | True   |
+    | "3.8"    | "3.8.1" | True   |
+    | "3"      | "3.8.1" | True   |
+    """
+    actual_version = platform.python_version()
+    return actual_version.startswith(expected_version)
+
+
+def create_env_if_not_exist(env_path):
+    """Creates a venv virtual environment"""
+    if not os.path.exists(env_path):
+        subprocess.call(["python3", "-m", "venv", env_path])
+        print("created" + str(env_path))
+        existed7 = False
+    else:
+        print(str(env_path) + " already exist")
+        existed7 = True
+    return existed7
+
+
+def install_piped_requirements(pip_path, requirements_path):
```

Comment:
What are "piped requirements"?

Reply:
Fixed: renamed it to `install_requirements_with_pip` for a better readability